src/webhook/github.js

```diff
@@ -73,54 +73,196 @@ module.exports = async (req, res) => {
         let author = await getOrCreateUser(prData.user.login);
         if (!author) return res.status(200).send('Failed to get or create user. Skipping process.');
 
-        if (action === 'opened') {
-            console.log(`PR #${prNumber} opened on main branch`);
+        const labels = prData.labels;
+        let highestPriorityLabel = labels.length > 0 ? labels.reduce((prev, current) => {
+            return (prPoints[current.name.toLowerCase()] > prPoints[prev.name.toLowerCase()]) ? current : prev;
+        }, labels[0]) : null;
 
-            await prisma.pullRequests.create({
-                data: {
-                    prNumber: prNumber,
-                    repository: req.body.repository.full_name,
-                    title: prData.title,
-                    state: 'open',
-                    url: prData.html_url,
-                    openedAt: new Date(prData.created_at),
-                    points: 0,
-                    authorId: author.githubId
+        if (action === 'synchronize'){
+            const existingPr = await prisma.pullRequests.findUnique({
+                where: {
+                    prNumber_repository: {
+                        prNumber: prNumber,
+                        repository: req.body.repository.full_name
+                    }
                 }
             });
+
+            if (existingPr && existingPr.state === 'merged') {
+                console.log(`PR #${prNumber} is already merged. Checking for label updates.`);
+                if (labels.length === 0 && existingPr.label !== "") {
+                    console.log(`No new labels given. Updating points to zero.`);
+                    await prisma.pullRequests.update({
+                        where: {
+                            prNumber_repository: {
+                                prNumber: prNumber,
+                                repository: req.body.repository.full_name
+                            }
+                        },
+                        data: {
+                            points: 0
+                        }
+                    });
+                    await prisma.users.update({
+                        where: { githubId: author.githubId },
+                        data: {
+                            points: { increment: 0 - existingPr.points }
+                        }
+                    });
+                    return res.status(200).send('PR merged and points updated if necessary.');
+                }
+                if (highestPriorityLabel) {
```

Comment:
now you made label optional in schema, so no need to add if here, without if will work, just do `highestPriorityLabel?.name`

Reply:
ok wait I think you can keep it here